quickwit-cli/tests/cli.rs

```diff
@@ -330,11 +335,91 @@ async fn test_all_with_s3_localstack() -> Result<()> {
         .read_exact(&mut data)
         .expect("Cannot read output");
     let process_output_str = String::from_utf8(data).unwrap();
-    let query_response =
-        reqwest::get("http://127.0.0.1:8182/api/v1/data1/search?query=title:modern")
-            .await?
-            .text()
-            .await?;
+    let query_response = reqwest::get(format!(
+        "http://127.0.0.1:8182/api/v1/{}/search?query=title:modern",
+        data_endpoint
+    ))
+    .await?
+    .text()
+    .await?;
+    server_process.kill().unwrap();
+
+    assert!(process_output_str.contains("http://127.0.0.1:8182"));
+    let result: Value =
+        serde_json::from_str(&query_response).expect("Couldn't deserialize response.");
+    assert_eq!(result["numHits"], Value::Number(Number::from(1i64)));
+
+    make_command(format!("delete --index-uri {} ", test_env.index_uri).as_str())
+        .assert()
+        .success();
+    let metadata_file_exist = object_storage.exists(Path::new("quickwit.json")).await?;
+    assert_eq!(metadata_file_exist, false);
+
+    Ok(())
+}
+
+#[tokio::test]
+#[serial]
+#[cfg_attr(not(feature = "ci-test"), ignore)]
+async fn test_all_with_s3_localstack_api() -> Result<()> {
```

Comment:
what is the difference between `test_all_with_s3_localstack_api` and `test_all_with_s3_localstack_cli` ?

Reply:
can you be more explicit about the difference between the two tests ?

Reply:
The api test is not completely switching to the api test for all commands. the idea is that one tests the cli and one tests the api via internal structs.

Reply:
I saw that afterwards, not sure if it's worth doing it as cli tests are quite heavy. That's ok for now but we need to keep in mind that to avoid inflating too much cli tests.